src/apps/activity_assignments/service.py

```diff
@@ -225,6 +304,69 @@ async def _get_assignments_entities(
 
         return entities
 
+    async def unassign_many(self, assignments_unassign: list[ActivityAssignmentDelete]) -> None:
+        """
+        Unassigns multiple activity assignments by marking them as deleted.
+
+        This method takes a list of assignment requests, retrieves the corresponding
+        assignment entities from the database, marks them as deleted, and then returns
+        the updated assignments.
+
+        Parameters:
+        -----------
+        assignments_unassign : list[ActivityAssignmentDelete]
+            A list of assignment creation objects that specify which assignments to unassign.
+            Each object contains details such as `activity_id`, `activity_flow_id`,
+            `respondent_subject_id`, and `target_subject_id`.
+
+        Returns:
+        --------
+        list[ActivityAssignment]
+            A list of `ActivityAssignment` objects that have been marked as deleted.
+        """
+        self._validate_unassignment(assignments_unassign)
+
+        activity_or_flow_ids = []
+        respondent_subject_ids = []
+        target_subject_ids = []
+
+        for assignment in assignments_unassign:
+            # Append only non-None values to the activity_or_flow_ids list
+            if assignment.activity_id is not None:
+                activity_or_flow_ids.append(assignment.activity_id)
+            elif assignment.activity_flow_id is not None:
+                activity_or_flow_ids.append(assignment.activity_flow_id)
+
+            # Append other necessary IDs
+            target_subject_ids.append(assignment.target_subject_id)
+            respondent_subject_ids.append(assignment.respondent_subject_id)
+
+        await ActivityAssigmentCRUD(self.session).unassign_many(
+            activity_or_flow_ids=activity_or_flow_ids,
+            respondent_subject_ids=respondent_subject_ids,
+            target_subject_ids=target_subject_ids,
+        )
+
+    def _validate_unassignment(self, assignments_unassign: list[ActivityAssignmentDelete]) -> None:
+        """
+        Validates that all mandatory parameters are filled and correct.
+
+        Raises specific validation errors for each type of validation failure.
+        """
+        for assignment in assignments_unassign:
```

Comment:
I think you are right, I guess I wasn't getting any feedback from the root_validator becuase I'm settiing 
 assignments_unassign: list[ActivityAssignmentDelete] 
 instead of using the class with the validator
 
I used the _validate function because I was using the foder delete endpoint as reference and we are not using root_validator function (we are not using classes in this scenario).

Reply:
But then I have a question. On test_assignment we calling the POST request and sending the information as a dictionary. Meaning that we are bypassing the root_validator: 

`        response = await client.post(
            self.activities_assignments_applet.format(applet_id=applet_one.id),
            data=dict(
                assignments=[
                    dict(
                        respondent_subject_id=tom_applet_one_subject.id,
                        target_subject_id=tom_applet_one_subject.id,
                    )
                ]
            ),
        )
`

and then we test if the request was correct. We retrieve the error from the create_many function at activity_assignments/service.py
            activity_or_flow_name: str = self._validate_assignment_and_get_activity_or_flow_name(assignment, entities)

at the service layer, similar to what I'm doing. This makes me thing that th two validations are required? 
@rcmerlo @mbanting

Reply:
@AlejandroCoronadoN using the example above, it looks like
`_validate_assignment_and_get_activity_or_flow_name` [verifies](https://github.com/ChildMindInstitute/mindlogger-backend-refactor/blob/feature/M2-7358-unassign-activity-or-flow-endpoint/src/apps/activity_assignments/service.py#L243-L263) that the `activity_id` or `activity_flow_id`, and `respondent_subject_id` and `target_subject_id` exist in the database.

This is different (ie. not a duplicate) than what the `root_validator` in `ActivityAssignmentCreate` is [checking for](https://github.com/ChildMindInstitute/mindlogger-backend-refactor/blob/feature/M2-7358-unassign-activity-or-flow-endpoint/src/apps/activity_assignments/domain/assignments.py#L23-L27), which is that one and only one of `activity_id` or `activity_flow_id` is provided

Reply:
Ok thanks for the clarification Marty. I just want to confirm. I need to delete the validation function and use the root-validator instead. Then I will need to change the tests creating wrong instances of the ActivitiesAssignmentDelete what will automatically catch the error.

Reply:
Ok, I'll wait for Rodrigo but it makes sense what you are saying. I'll be working on the other tickets in the meantime. Thanks!

Reply:
we don't need theses validation pydantic already doing them in ActivityAssignmentDelete class

about your comments that we are sending a dict bypassing pydantic validators, it is not true, because fastapi will always receive a JSON, transform it to ActivityAssignmentDelete or ActivityAssignmentCreate classes and it will trigger pydantic validators

in some tests we are creating ActivityAssignmentCreate classes directly, to be easier to create the tests, but at the end http client will transform everything to JSON

Test it manually directly in postman to verify

Reply:
Thanks for you comments Rodrigo, I applied the changes as you suggested. What confused me is that when I run:
```
        assignment_delete = ActivitiesAssignmentsDelete(
            assignments=[
                ActivityAssignmentDelete(
                    activity_flow_id=str(applet_one.activities[0].id),
                    respondent_subject_id=str(tom_applet_one_subject.id),
                    target_subject_id=None,
                )
            ]
        )
```
        
      The error happens even before reaching 
              unassign_response = await client.delete(...) 
              
 This is because when building the class the root validator is executed and we retyurn an error, however, when creating the test this error is not returned as a response and I was unable to compare it aor create and assert operation. So instead we create a dictionary: 
 
 `
         assignment_delete = {
            "assignments": [
                {
                    "activity_id": str(applet_one.activities[0].id),
                    "respondent_subject_id": str(tom_applet_one_subject.id),
                    "target_subject_id": None,  # Missing target_subject_id
                }
            ]
        }

 `
 
 This make me think that we were avoiding the error by passing a dictionary, but this is correct. As you metioned the error is eventually raise by the cliente.delete() request and that way I'm able to  test my response. 
 
 I just had to delete the _validation function and let the root_validator handle all the errors.